dashboard/src/main/java/org/openedx/learn/presentation/LearnViewModel.kt

```diff
@@ -1,19 +1,39 @@
 package org.openedx.learn.presentation
 
 import androidx.fragment.app.FragmentManager
+import androidx.lifecycle.viewModelScope
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.update
+import kotlinx.coroutines.launch
 import org.openedx.DashboardNavigator
 import org.openedx.core.config.Config
 import org.openedx.dashboard.presentation.DashboardAnalytics
 import org.openedx.dashboard.presentation.DashboardAnalyticsEvent
 import org.openedx.dashboard.presentation.DashboardAnalyticsKey
 import org.openedx.dashboard.presentation.DashboardRouter
 import org.openedx.foundation.presentation.BaseViewModel
+import org.openedx.learn.LearnType
 
 class LearnViewModel(
+    openTab: String,
     private val config: Config,
     private val dashboardRouter: DashboardRouter,
     private val analytics: DashboardAnalytics,
 ) : BaseViewModel() {
+    private val _uiState = MutableStateFlow(
+        LearnUIState(
+            if (openTab == LearnTab.PROGRAMS.name) {
```

Comment:
What do you think about using valueOf to convert a string to an enum instead of using if?